### PR TITLE
Basic raster working copy: diffs and commits

### PR DESCRIPTION
**Pull request by olsen232**
## Description

Allows for the displaying and committing uncommitted changes into raster.v1 datasets.
Note that raster.v1 datasets are still "geotiffs, may or may not be COGs" - implementing COG-only datasets will make this code a bit more complex as it will mean converting to COG during a diff / commit.

## Related links:

https://github.com/koordinates/kart/issues/794

## Checklist:

- [x] Have you reviewed your own change?
- [x] Have you included test(s)?
- [ ] Have you updated the [changelog](https://github.com/koordinates/kart/blob/master/CHANGELOG.md)?

### CODE REVIEW

File: kart/point_cloud/tilename_util.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def get_tile_path_pattern(
     parent_pattern = (
         re.escape(parent_path.rstrip("/") + "/") if parent_path is not None else ""
     )
-    tile_pattern = re.escape(tilename) if tilename is not None else r"([^/]+)"
+    tile_pattern = re.escape(tilename) if tilename is not None else r"([^/.][^/]*)"
```
Comment (craigds) [inline kart/point_cloud/tilename_util.py:54]: why this change?

File: kart/raster/v1.py [hunk 2]
```diff
@@ -26,3 +32,103 @@ def set_tile_extension(cls, filename, ext=None, tile_format=None):
...
+            include_conflict_versions=include_conflict_versions,
+        )
+
+    def diff_to_working_copy(
+        self,
+        workdir_diff_cache,
+        ds_filter=DatasetKeyFilter.MATCH_ALL,
+        *,
+        convert_to_dataset_format=False,
+        skip_gdalinfo=False,
```
Comment (craigds) [inline kart/raster/v1.py:52]: minor: positive flags are easier to reason about (`use_gdalinfo=True`)

File: kart/schema.py [hunk 3]
```diff
@@ -227,8 +227,22 @@ def name(self):
         return self.get("name")
 
     @property
-    def id_or_name(self):
-        return self.get("id") if "id" in self else self.get("name")
+    def interpretation(self):
+        return self.get("interpretation")
```
Comment (craigds) [inline kart/schema.py:231]: what is this? maybe add a docstring

**Review by craigds — CHANGES_REQUESTED** [review]

File: tests/raster/fixtures.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+from kart import is_darwin
+
+# using a fixture instead of a skipif decorator means we get one aggregated skip
+# message rather than one per test
+@pytest.fixture(scope="session")
+def requires_gdal_info(data_archive_readonly):
+    from osgeo import gdal
+
+    # TODO - figure out what's wrong with gdal.Info in our macos build.
```
[the remark this answers is not in the document]
Comment (olsen232) [reply tests/raster/fixtures.py:14]: Done

File: kart/tile/tilename_util.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from kart.point_cloud.tilename_util import remove_tile_extension as remove_pc_extension
+from kart.point_cloud.tilename_util import (
+    remove_tile_extension as remove_raster_extension,
+)
```
[the remark this answers is not in the document]
Comment (olsen232) [reply kart/tile/tilename_util.py:4]: Oops! Fixed.
Also tightened up the code to not remove both types of extension at once eg .copc.laz.tif

File: kart/schema.py (hunk 3 above)
Comment (olsen232) [reply kart/schema.py:231]: Done

File: kart/point_cloud/tilename_util.py (hunk 1 above)
Comment (olsen232) [reply kart/point_cloud/tilename_util.py:54]: Gdal creates temporary, hidden-ish files sometimes eg
aerial.tif -> ._aerial.tif
I had to make sure the raster WC would ignore such files to get the tests to pass - and I made the point-cloud WC ignore such hidden-ish files too for consistency

File: kart/raster/v1.py (hunk 2 above)
Comment (olsen232) [reply kart/raster/v1.py:52]: Renamed to `extract_metadata`, and the same for point-clouds.
Possibly, this fixed some functionality for rasters that I haven't yet looked into involving kart reset. In further PRs I will need to add more tests to make sure rasters is at feature parity with point-clouds and find out where it isn't

File: kart/point_cloud/tilename_util.py (hunk 1 above)
Comment (craigds) [reply kart/point_cloud/tilename_util.py:55]: ok. i'd suggest adding that as a comment here. it's not otherwise clear why this regex ignores files starting with a `.` - it might come back to bite us (esp on windows where those files aren't considered 'hidden' necessarily)

**Review by craigds — APPROVED** [review]

File: kart/point_cloud/tilename_util.py (hunk 1 above)
Comment (olsen232) [reply kart/point_cloud/tilename_util.py:55]: Done
